### ISSUE TITLE
dotnet host should log fatal application errors to Windows Event Log (when on Windows)

### ISSUE THREAD

**Issue by jkotas** [dotnet/coreclr#16735]
_From @livarcocc on March 2, 2018 22:29_

_From @DamianEdwards on March 2, 2018 22:22_

.NET Framework on Windows today will log an error from the source ".NET Runtime", including the offending stack trace and application name, to the "Application" Windows Event Log whenever a .NET app crashes, e.g.:

```
Application: VSIXAutoUpdate.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.ArgumentNullException
   at Microsoft.VisualStudio.Telemetry.TelemetryComplexProperty..ctor(System.Object)
   at VSIXAutoUpdate.AutoUpdate.LogAutoUpdateCompletedTelemetry()
   at VSIXAutoUpdate.AutoUpdate.Main()

```

![image](https://user-images.githubusercontent.com/249088/36924946-7e797768-1e25-11e8-907a-7dd92f560d29.png)

This information is very useful in many scenarios, including servers, where many existing systems collect Windows Event Log information and use it to report and diagnose problem applications.

We'd very much like this feature to be replicated in the .NET Core host for Windows, such that any faulting .NET Core application logs a similar error.

@davidfowl 

_Copied from original issue: dotnet/cli#8719_

_Copied from original issue: dotnet/core-setup#3765_

**Comment by jkotas** [comment]
_From @livarcocc on March 2, 2018 22:29_

I will move this to core-setup, where the dotnet host lives.

**Comment by jkotas** [comment]
_From @DamianEdwards on March 2, 2018 23:27_

@livarcocc do you have any idea what this would take? I'm assuming we could just re-use the existing ".NET Runtime" source so that no install requirements are added.

**Comment by jkotas** [comment]
_From @davidfowl on March 3, 2018 16:27_

/cc @jkotas @richlander @Petermarcu

**Comment by jkotas** [comment]
The fix should be to delete `FEATURE_CORESYSTEM` here: https://github.com/dotnet/coreclr/blob/67f40646d76d914ab7f099df25768c73c4f55bad/src/vm/eventreporter.cpp#L451 and verify that it works on both regular Windows and NanoServer. This ifdef is there because of the original "CoreSystem" reduced surface Windows editions did not contain eventlog.

**Comment by jkotas** [comment]
Related #9529 and #14037

cc @sywhang

**Comment by davidfowl** [comment]
It's just an ifdef 😄

**Comment by DamianEdwards** [comment]
OMG :shipit: already

**Comment by sywhang** [comment]
This should be now fixed with #16875
